pkg/volume/iscsi/iscsi_util.go

```diff
@@ -390,3 +407,31 @@ func extractPortalAndIqn(device string) (string, string, error) {
 	iqn := device[ind2:ind]
 	return portal, iqn, nil
 }
+
+func removeDuplicatePortal(bkpPortal []string) []string {
+	portals := make([]string, 0, len(bkpPortal))
+	found := map[string]bool{}
+	for _, tp := range bkpPortal {
+		if !found[tp] {
+			found[tp] = true
+			portals = append(portals, tp)
+		}
+	}
+	return portals
```

Comment:
Thank you for your comment. Your suggestion looks good.
My go is not great too, so I'm curious what kind of input causes empty elements return?